### PR TITLE
Feature/261/Add available statuses field to View all orders endpoint

### PR DESCRIPTION
**Pull request by ABbondar**
## Issues
#261 
## Summary of changes🔥
- Add available statuses field to View all orders endpoint.
- Implemented tests to cover all new lines of code.

### CODE REVIEW

**Review by ryhaldenys — DISMISSED** [review]

**Review by urio999 — DISMISSED** [review]

File: code/orders-api-rest/api/openapi-rest.yml [hunk 1]
```diff
@@ -1056,6 +1056,12 @@ components:
           description: Indicates whether the order has been paid
         orderStatus:
           $ref: '#/components/schemas/OrderStatus'
+        availableStatuses:
+          type: array
+          items:
+            type: string
```
Comment (DenysLiubchenko) [inline code/orders-api-rest/api/openapi-rest.yml]: I think you can use OrderStatus component

**Review by DenysLiubchenko — CHANGES_REQUESTED** [review]

File: code/orders-application/src/main/java/com/academy/orders/application/order/usecase/CalculateOrderTotalPriceUseCaseImpl.java [hunk 2]
```diff
@@ -34,4 +41,31 @@ public List<Order> calculateTotalPriceFor(List<Order> orders) {
...
 	}
+
+	@Override
+	public List<OrderManagement> calculateTotalPriceAndAvailableStatuses(List<Order> orders,
+			String currentAccountEmail) {
+		if (orders == null) {
+			return Collections.emptyList();
+		}
+
+		var accountRole = accountRepository.findRoleByEmail(currentAccountEmail)
```
Comment (DenysLiubchenko) [inline code/orders-application/src/main/java/com/academy/orders/application/order/usecase/CalculateOrderTotalPriceUseCaseImpl.java]: You can get authorities from Authentication and avoid doing request into db

Comment (ABbondar) [reply code/orders-application/src/main/java/com/academy/orders/application/order/usecase/CalculateOrderTotalPriceUseCaseImpl.java]: Ok, changed

File: code/orders-api-rest/api/openapi-rest.yml (hunk 1 above)
Comment (ABbondar) [reply code/orders-api-rest/api/openapi-rest.yml]: Yes, i will reuse this component

**Review by DenysLiubchenko — APPROVED** [review]

**Pull request merged**